system-test/tpc-tests.ts

```diff
@@ -0,0 +1,112 @@
+import {Bigtable} from '../src';
+
+describe.skip('TPC tests', () => {
+  // These tests are only designed to pass when using the service account
+  // credentials for the TPC environment so we skip them in the CI pipeline.
+  //
+  // To see successful tests, uncomment the following line:
+  // process.env.GOOGLE_APPLICATION_CREDENTIALS = '/path-to/tpc_sa_key.json';
+
+  function runTest(done: Mocha.Done, bigtable: Bigtable) {
+    (async () => {
+      try {
+        const instance = bigtable.instance(instanceId);
+        const table = instance.table(tableId);
+
+        // Create the instance if it doesn't exist:
+        const [instanceInfo] = await instance.exists();
+        if (!instanceInfo) {
+          const [, operation] = await instance.create({
+            clusters: {
+              id: 'fake-cluster3',
+              location: 'u-us-prp1-a',
+              nodes: 1,
+            },
+          });
+          await operation.promise();
+        }
+
+        // Create the table and the column families.
+        const [tableExists] = await table.exists();
+        if (!tableExists) {
+          await table.create({families: [columnFamilyId]}); // Create column family
+        } else {
+          // Check if column family exists and create it if not.
+          const [families] = await table.getFamilies();
+
+          if (
+            !families.some(
+              (family: {id: string}) => family.id === columnFamilyId,
+            )
+          ) {
+            await table.createFamily(columnFamilyId);
+          }
+        }
+
+        // Run the readrows operation:
+        await table.getRows();
+        done();
+      } catch (e) {
+        done(e);
+      }
+    })().catch(err => {
+      throw err;
+    });
+  }
+
+  const instanceId = 'emulator-test-instance';
+  const tableId = 'my-table';
+  const columnFamilyId = 'cf1';
+
+  after(async () => {
+    // TODO: Solve the issue where tests fail because instances don't get created on time.
+    // Notes: Creating instances can take time and if they are not ready in
+    // time then tests can fail. This shouldn't happen because if the create
+    // instance long running operation completes then the instance should be
+    // ready and shouldn't produce the `Error: 5 NOT_FOUND` error.
+    // Uncomment the code below when the task above is addressed:
+    // const instance = bigtable.instance(instanceId);
+    // await instance.delete({});
+  });
+
+  it('should set the TPC universe with a client option', done => {
+    const universeDomain = 'apis-tpczero.goog'; // or your universe domain if not using emulator
+    const options = {
+      universeDomain,
+    };
+    const bigtable = new Bigtable(options);
+    runTest(done, bigtable);
+  });
+
+  it('should set the TPC universe with a client option for the gapic clients', done => {
+    const universeDomain = 'apis-tpczero.goog'; // or your universe domain if not using emulator
+    const options = {
```

Comment:
I know basically nothing at all about bigtable libraries, but it seems odd to me that each of the client has its own options variants. Maybe something we think about later, if they could be combined.

Reply:
This is just one potential option. The user can pass universe domain in here for each individual Gapic client or like one of the other tests they can just pass in the universeDomain for the handwritten client and that universe domain option will be applied to all three. So the combined option is supported.

Do we want to remove this test and eliminate the support for universe domain on the Gapic clients?

Reply:
Ah, I see. Thanks for the clarification.